diff --git a/pom.xml b/pom.xml
index 8a3d6f1..6d148ff 100644
--- a/pom.xml
+++ b/pom.xml
@@ -209,10 +209,12 @@
           </execution>
         </executions>
       </plugin>
+<!--
       <plugin>
         <groupId>org.graalvm.buildtools</groupId>
         <artifactId>native-maven-plugin</artifactId>
       </plugin>
+-->
       <plugin>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-maven-plugin</artifactId>
@@ -230,8 +232,20 @@
                 <java.source>${java.version}</java.source>
                 <java.target>${java.version}</java.target>
               </additionalProperties>
+              <image>
+                <builder>paketobuildpacks/builder:tiny</builder>
+                <env>
+                  <BP_NATIVE_IMAGE>true</BP_NATIVE_IMAGE>
+                </env>
+              </image>
             </configuration>
           </execution>
+          <execution>
+            <id>process-aot</id>
+            <goals>
+              <goal>process-aot</goal>
+            </goals>
+          </execution>
         </executions>
       </plugin>
       <plugin>
diff --git a/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java b/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
index ac6e150..fe5c747 100644
--- a/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
+++ b/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
@@ -16,6 +16,9 @@
 
 package org.springframework.samples.petclinic;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.ImportRuntimeHints;
@@ -31,7 +34,26 @@ import org.springframework.context.annotation.ImportRuntimeHints;
 public class PetClinicApplication {
 
 	public static void main(String[] args) {
+		long mainStart = System.currentTimeMillis();
+		RuntimeMXBean runtimeMXBean = ManagementFactory.getRuntimeMXBean();
+		// This includes all the time spent inside the JVM before main() is reached
+		// (since os::Posix::init is called and initial_time_count is initialized).
+		long vmStart = runtimeMXBean.getStartTime();
+		long maxBeanOverHead = System.currentTimeMillis() - mainStart;
+
 		SpringApplication.run(PetClinicApplication.class, args);
+		long end = System.currentTimeMillis();
+		Boolean quit = Boolean.getBoolean("autoQuit");
+		if (quit) {
+			System.out.println("#### Booted and returned in " + (end - vmStart - maxBeanOverHead) + "ms");
+			System.out.println("#### (debug) mainStart = " + mainStart);
+			System.out.println("#### (debug) vmStart = " + vmStart);
+			System.out.println("#### (debug) before main (mainStart - vmStart) = " + (mainStart - vmStart));
+			System.out.println("#### (debug) maxBeanOverHead = " + maxBeanOverHead);
+			System.out.println("#### (debug) end = " + end);
+			System.exit(0);
+		}
+
 	}
 
 }
